source/type/tests/test.cpp

```diff
@@ -216,3 +218,69 @@ BOOST_AUTO_TEST_CASE(tz_handler_overflow_test) {
     BOOST_CHECK_EQUAL_RANGE(periods, build_dst_periods);
     BOOST_CHECK_EQUAL(build_dst_periods.begin()->first, 60*60*12);
 }
+
+BOOST_AUTO_TEST_CASE(get_sections_stop_points){
+    ed::builder b("20120614");
+    // 0 1 0 2 3 0 4 2
+    b.vj("A")
+        ("0", 8000, 8000)
+        ("1", 8000, 8000)
+        ("0", 8000, 8000)
+        ("2", 8000, 8000)
+        ("3", 8000, 8000)
+        ("0", 8000, 8000)
+        ("4", 8000, 8000)
+        ("2", 8000, 8000)
+        ("5", 8000, 8000)
+        ("2", 8000, 8000);
+    b.data->pt_data->index();
+    b.data->complete();
+    b.finish();
+
+    const auto* vj = b.data->pt_data->vehicle_journeys.at(0);
+    const auto sa = std::vector<type::StopArea*>{
+        b.data->pt_data->stop_areas_map.at("0"),
+        b.data->pt_data->stop_areas_map.at("1"),
+        b.data->pt_data->stop_areas_map.at("2"),
+        b.data->pt_data->stop_areas_map.at("3"),
+        b.data->pt_data->stop_areas_map.at("4"),
+        b.data->pt_data->stop_areas_map.at("5"),
+    };
+    const auto sp = std::vector<type::StopPoint*>{
```

Comment:
completly minor too, but you can save lines by making it a lamba
`auto sp = [&](const std::string& id) { return b.get<nt::StopPoint>(id); }` (not tested)